src/lib/log-forwarding.js

```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const rtLib = require('@adobe/aio-lib-runtime')
+const { writeAio, writeEnv } = require('./import')
+const crypto = require('crypto')
+const fs = require('fs-extra')
+
+const SECRET_FIELD_TYPE = 'password'
+const CHECKSUM_FILE = 'log-forwarding-config.sha256'
+
+class LogForwarding {
+  constructor (aioConfig) {
+    this.aioConfig = aioConfig
+  }
+
+  async init () {
+    rtLib.utils.checkOpenWhiskCredentials({ ow: this.aioConfig.runtime })
+    this.logForwarding = await getRTLogForwarding(this.aioConfig.runtime)
+    return this
+  }
+
+  getLocalConfig () {
+    const config = this.aioConfig.project.workspace.log_forwarding
+    try {
+      return convertToConfigObject(config)
+    } catch (e) {
+      throw new Error('Incorrect local log forwarding configuration. ' + e.message)
+    }
+  }
+
+  getLocalConfigWithSecrets () {
+    const config = this.getLocalConfig()
+    const destination = config.getDestination()
+    const settings = config.getSettings()
+    if (config.isDefined()) {
+      const destinationSettings = this.logForwarding.getDestinationSettings(destination)
+      const missingSecrets = []
+      destinationSettings.forEach(e => {
+        if (e.type === SECRET_FIELD_TYPE) {
+          const secretVarName = getSecretVarName(destination, e.name)
+          if (process.env[secretVarName] !== undefined) {
+            settings[e.name] = process.env[secretVarName]
+          } else {
+            missingSecrets.push(secretVarName)
+          }
+        }
+      })
+      if (missingSecrets.length > 0) {
+        throw new Error('Required secrets are missing in environment variables: ' + missingSecrets.join(', ') + '. ' +
+          'Make sure these variables are set in .env file')
+      }
+    }
+    return new LogForwardingConfig(destination, settings)
+  }
+
+  async getServerConfig () {
+    try {
+      return convertToConfigObject(await this.logForwarding.get())
+    } catch (e) {
+      throw new Error('Incorrect log forwarding configuration on server. ' + e.message)
+    }
+  }
+
+  getSupportedDestinations () {
```

Comment:
nit: 1 line becomes 3 lines, just call this.logForwarding.getSupportedDestinations() directly

Reply:
Yes. I was thinking about those two ways. The only reason for the current implementation is fewer dependencies in the actual command. Right now, it depends only on this library. If we remove this method, it will have to depend also on the RT LogForwarding directly. I chose one of the options as it didn't look critical for me and this library looks just like a wrapper. But if it's a concern, I can remove these two methods.

Reply:
my mistake, didn't see that this was called from outside